sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -4101,6 +4115,152 @@ private void SetNameFieldsIfNull()
             }
         }
         #endregion
+
+        #region GenerateSAS
+        /// <summary>
+        /// The <see cref="GetSasBuilder"/> returns a <see cref="BlobSasBuilder"/> that
+        /// sets the respective properties in the BlobSasBuilder from the client.
+        /// </summary>
+        /// <param name="permissions">
+        /// Specifies the list of permissions that can be set in the SasBuilder
+        /// See <see cref="BlobSasPermissions"/>.
+        /// </param>
+        /// <param name="expiresOn">
+        /// Specifies when to set the expires time in the sas builder
+        /// </param>
+        /// <returns>
+        /// A <see cref="BlobSasBuilder"/> on successfully deleting.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public BlobSasBuilder GetSasBuilder(
+            BlobSasPermissions permissions,
+            DateTimeOffset expiresOn)
```

Comment:
This kind of duplicates functionality of the builder as it has methods to set these. The goal of this API is to inject credentials and other properties derived from blob client rather than exposing api mirroring builder.
Should we make this parameterless?

Reply:
We chose these as the parameters cause they are both required to construct the sas, the other fields are optional. I think we set these as the parameters, to avoid a user error and to show what is required to construct a valid sas.

Reply:
that makes sense. thanks.

Reply:
Well you could also just set the identifier in SAS, so this isnt the only minimum set of required parameters

Reply:
Signed identifiers require a lot more setup though and I think it's a more advanced scenario.  These are very much about letting customers know that a working client plus these two values are all you need to generate a minimalist SAS token for the blob you just uploaded that you can then hand off to another service.